subsys/caf/modules/sensor_data_aggregator.c

```diff
@@ -20,68 +20,72 @@ LOG_MODULE_REGISTER(MODULE, CONFIG_CAF_SENSOR_DATA_AGGREGATOR_LOG_LEVEL);
 
 #define DT_DRV_COMPAT caf_aggregator
 
-#define __DEFINE_DATA(i, agg_id, size)							\
+#define __DEFINE_DATA(i, agg_id, size)								\
 	static uint8_t agg_ ## agg_id ## _buff_ ## i ## _data[size] __aligned(4)
 
 #if (DT_INST_NODE_HAS_PROP(agg_id, memory_region))
 
-#define __INITIALIZE_BUF(i, agg_id)							\
-	{										\
-		(uint8_t *) DT_REG_ADDR(DT_INST_PHANDLE(agg_id, memory_region)) +	\
-		i * DT_INST_PROP(agg_id, buf_data_length)				\
+#define __INITIALIZE_BUF(i, agg_id)								\
+	{											\
+		(struct sensor_value *) DT_REG_ADDR(DT_INST_PHANDLE(agg_id, memory_region)) +	\
+		i * ((DT_INST_PROP(agg_id, buf_data_length)))					\
 	},
 
 
 
-/* buf_len has to be equal to n*sensor_data_size. */
+/* buf_len has to be equal to i*sample_size*sizseof(struct sensor_value). */
 #define __DEFINE_BUF_DATA(i)									\
 	static struct aggregator_buffer agg_ ## i ## _bufs[] = {				\
 		UTIL_LISTIFY(DT_INST_PROP(i, buf_count), __INITIALIZE_BUF, i)			\
 	};											\
-	BUILD_ASSERT((DT_INST_PROP(i, buf_data_length) % DT_INST_PROP(i, sensor_data_size)) == 0,\
+	BUILD_ASSERT((DT_INST_PROP(i, buf_data_length) %					\
```

Comment:
Doesn't code lowers requirement we are going to check her?

Reply:
Discussed offline. Conclusion: keep it.